pkg/operator/ceph/disruption/nodedrain/reconcile.go

```diff
@@ -82,6 +83,18 @@ func (r *ReconcileNode) reconcile(request reconcile.Request) (reconcile.Result,
 		return reconcile.Result{}, fmt.Errorf("Label key %s does not exist on node %s", corev1.LabelHostname, request.NamespacedName)
 	}
 
+	osdPodList := &corev1.PodList{}
+	err = r.client.List(context.TODO(), osdPodList, client.MatchingLabels{k8sutil.AppAttr: osd.AppName})
```

Comment:
If there are a lot of nodes and a lot of OSDs, this could be slow if this reconciler is triggered very often. How noisy could it be?

Reply:
Here's how it works internally:

A watch is registered to ensure the cache reflects the state of the objects, create/update/delete events are generated from the cache, a predicate is applied on the update event to ensure only updates to the Pod's node name enqueue requests, the event handler enqueues requests for nodes based on the node name and the queue is automatically deduplicated.

The result is that one worker goroutine will run one instance of Reconcile(nodeName) for each node that either an OSD left or an OSD started on.

So the noise is filtered out at the predicate and event handler level. I'm exploring how to use informer indexes and list watches filtered on the server side for future improvements.

I do think that this won't be too noisy though.

Reply:
Realized you're probably talking about listing OSD pods in every reconcile loop, but the List operation here is done on the cache, so that should be fine too. Building an Index of node to OSD in the cache would be computationally more efficient, but I can't get controller-runtime to play nicely with that yet.

This does not affect the amount API calls made as the only api call is the watch stream that populates the cache.